sql/src/main/java/io/crate/analyze/relations/RelationSplitter.java

```diff
@@ -121,10 +123,12 @@ private static void splitOrderBy(AnalyzedRelation analyzedRelation,
         SplitContext splitContext = splitForRelation(analyzedRelation, orderBy.orderBySymbols());
         addAllNew(splitOutputs, splitContext.mixedSplit);
 
+        OrderBy remainingOrderBy = orderBy;
         if (!splitContext.directSplit.isEmpty()) {
-            rewriteOrderBy(querySpec, splitQuerySpec, splitContext);
+            remainingOrderBy = rewriteOrderBy(remainingOrderBy, splitQuerySpec, splitContext);
```

Comment:
pls check that rewriteOrderBy uses a copying visitor. otherwise normalizing the returned order by might change the other